tests/unit/moduleapi/crash.tcl

```diff
@@ -0,0 +1,50 @@
+# This file is used to test certain crash edge cases to make sure they produce
+# correct stack traces for debugging.
+set testmodule [file normalize tests/modules/crash.so]
+
+# Valgrind will complain that the process terminated by a signal, skip it.
+if {!$::valgrind} {
+    start_server {tags {"modules"}} {
+        r module load $testmodule assert
+        test {Test module crash when info crashes with an assertion } {
+            catch {r 0 info infocrash}
+            set res [wait_for_log_messages 0 {"*=== REDIS BUG REPORT START: Cut & paste starting from here ===*"} 0 10 1000]
+            set loglines [lindex $res 1]
+
+            set res [wait_for_log_messages 0 {"*ASSERTION FAILED*"} $loglines 10 1000]
+            set loglines [lindex $res 1]
+
+            set res [wait_for_log_messages 0 {"*RECURSIVE ASSERTION FAILED*"} $loglines 10 1000]
+            set loglines [lindex $res 1]
+
+            wait_for_log_messages 0 {"*=== REDIS BUG REPORT END. Make sure to include from START to END. ===*"} $loglines 10 1000
+            assert_equal 1 [count_log_message 0 "=== REDIS BUG REPORT END. Make sure to include from START to END. ==="]
+            assert_equal 2 [count_log_message 0 "ASSERTION FAILED"]
```

Comment:
let's also make sure we see the stack trace (check that `assertCrash` is mentioned), twice.

Reply:
It's technically three times, but sure.

Reply:
why 3 times?

also, despite already merged, can't / shouldn't we make a similar validation that the nested stack trace works for the segfault test too?

Reply:
The second assertion still prints the stack trace, so the crash will be in the stack trace twice. 

> also, despite already merged, can't / shouldn't we make a similar validation that the nested stack trace works for the segfault test too?

I didn't follow this.

Reply:
> The second assertion still prints the stack trace, so the crash will be in the stack trace twice.

right, so why is it 3 times?

> > also, despite already merged, can't / shouldn't we make a similar validation that the nested stack trace works for the segfault test too?
> 
> I didn't follow this.

we have two tests, one "with an assertion" and one "with a segfault".
the first one checks that the `assertCrash` function appears in both the original stack trace, and the nested one (and a 3rd?).
but the other test, only verifies that we show `Crashed running the instruction at` twice, and it doesn't validates that the stack trace is printed twice too.

Reply:
https://github.com/redis/redis/pull/12915